src/uwtools/cli.py

```diff
@@ -305,6 +306,121 @@ def _dispatch_forecast_run(args: Namespace) -> bool:
     )
 
 
+# Mode Rocoto
+
+
+def _add_subparser_rocoto(subparsers: Subparsers) -> ModeChecks:
+    """
+    Subparser for mode: rocoto
+
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    """
+    parser = _add_subparser(subparsers, STR.rocoto, "Write and validate Rocoto XML Documents")
+    _basic_setup(parser)
+    subparsers = _add_subparsers(parser, STR.submode)
+    return {
+        STR.write: _add_subparser_rocoto_write(subparsers),
+        STR.validate: _add_subparser_rocoto_validate(subparsers),
+    }
+
+
+def _add_subparser_rocoto_write(subparsers: Subparsers) -> SubmodeChecks:
+    """
+    Subparser for mode: rocoto write
+
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    """
+    parser = _add_subparser(subparsers, STR.write, "Write a Rocoto XML workflow document")
+    required = parser.add_argument_group(TITLE_REQ_ARG)
+    _add_arg_input_file(required)
+    _add_arg_schema_file(required)
+    _add_arg_output_file(required)
+    optional = _basic_setup(parser)
+    _add_arg_input_format(optional, choices=[FORMAT.yaml])
+    _add_arg_output_format(optional, choices=[FORMAT.jinja2])
+    checks = _add_args_quiet_and_verbose(optional)
+    return checks + [
+        partial(_check_file_vs_format, STR.infile, STR.infmt),
+        partial(_check_file_vs_format, STR.outfile, STR.outfmt),
+    ]
+
+
+def _add_subparser_rocoto_validate(subparsers: Subparsers) -> SubmodeChecks:
+    """
+    Subparser for mode: rocoto validate
+
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    """
+    parser = _add_subparser(subparsers, STR.validate, "Validate Rocoto XML")
+    required = parser.add_argument_group(TITLE_REQ_ARG)
+    _add_arg_input_file(required)
+    optional = _basic_setup(parser)
+    _add_arg_input_format(optional, choices=[FORMAT.jinja2])
+    checks = _add_args_quiet_and_verbose(optional)
+    return checks + [
+        partial(_check_file_vs_format, STR.infile, STR.infmt),
+    ]
+
+
+def _dispatch_rocoto(args: Namespace) -> bool:
+    """
+    Dispatch logic for rocoto mode.
+
+    :param args: Parsed command-line args.
+    """
+    return {
+        STR.write: _dispatch_rocoto_write,
+        STR.validate: _dispatch_rocoto_validate,
+    }[
+        args.submode
+    ](args)
+
+
+def _dispatch_rocoto_write(args: Namespace) -> bool:
+    """
+    Dispatch logic for rocoto write submode. Validate input and output.
+
+    :param args: Parsed command-line args.
+    """
+    success = True
+    if args.input_format == FORMAT.yaml and args.output_format == FORMAT.jinja2:
+        valid_input = uwtools.config.validator.validate_yaml(
+            config_file=args.input_file, schema_file=args.schema_file
+        )
+        if valid_input:
```

Comment:
This might clean up with a walrus operator:

```
if not valid_input := uwtools.config.validator.validate_yaml(
            config_file=args.input_file, schema_file=args.schema_file
        ):
    return valid_input

uwtools.rocoto.write_rocoto_xml(....)
```

Reply:
My head is still in 3.7 I think, I always forget about that! I ended up removing the entire `valid_input` logic though as there was no need to use it